theano/gof/cmodule.py

```diff
@@ -2000,6 +2025,20 @@ def join_options(init_part):
                     _logger.info("g++ -march=native equivalent flags: %s",
                                  GCC_compiler.march_flags)
 
+            # Find working march flag:
+            default_detected_flag = GCC_compiler.march_flags[0][7:]
```

Comment:
if not try_march_flag(GCC_compiler.march_flags):
   for march_flag in ['corei7-avx', 'corei7', 'core2']:
      GCC_compiler.march_flags = march_flags
      if try_march_flags(march_flag):
         break